packages/fxa-react/components/Portal/index.tsx

```diff
@@ -1,6 +1,8 @@
 import React, { memo, useEffect } from 'react';
 import { createPortal } from 'react-dom';
 
+import '../../styles/tailwind.out.css';
```

Comment:
Hmm, importing this for every `fxa-react` component would not be fun. 😩 In our `tailwind.config.js` files, we add `fxa-react` components to the configs with `config.content.push` and get them for production via `extractImportedComponents` but I know that only looks for Tailwind classes.

I wonder if for `fxa-react`, we should import CSS files directly like importing `portal.css` here instead of `tailwind.out.css`, but the problem with that is when we use `@apply` in CSS files in there, it would need to get compiled first. I think ideally we should somehow be including `fxa-react/styles/tailwind.css` somewhere when compiling other packages, but especially if `fxa-react` grew of course we wouldn't want classes from other components in there that we aren't using, almost like we need a `extractImportedComponents` for CSS files.

I think if there was a way to not have to remember to import the `.out` file every time a component has its own `.css` file, this would feel better. That's not necessarily blocking r+ though (maybe with a follow up) if this is causing issues for end users but I don't know what this banner is supposed to look like from the Jira ticket since the screenshot is without the banner showing.

Btw is this an issue in production + dev + storybook?

Reply:
As of yesterday this issue is now in Production. The banner is displayed, it's just popping up at the bottom of the page, so you won't see it unless you scroll down, since it loads at +100vh. (screenshot below)

Maybe something I want to check regarding `extractImportedComponents`. Is the idea that in Prod we don't use the `tailwind.out.css` from `fxa-react` and instead the packages using `fxa-react` components include the tailwind classes in their `tailwind.out.css`?

Another question/suggestion, could we include a root `index.ts` and maybe even `index.css` files in `fxa-react`? This way we import `tailwind.out.css` once, and we are good to go.

Another question, once we start including custom tailwind utility classes in `fxa-react` won't we need to include the `tailwind.out.css` from `fxa-react` in any case?

Reply:
> Maybe something I want to check regarding `extractImportedComponents`. Is the idea that in Prod we don't use the `tailwind.out.css` from `fxa-react` and instead the packages using `fxa-react` components include the tailwind classes in their `tailwind.out.css`?

Yeah, that happens right now in prod and locally. It looks at what components are imported and adds those components to the Tailwind "content" config option to tell it to look for Tailwind classes in those components. I think it would be the most ideal to get it to somehow include any relevant CSS files from those components but I'm not sure if the Tailwind config has an option like that.

Hmm, just thought of something that might be slick. Would it be possible to import `fxa-react`'s `tailwind.css` file into each package's `tailwind.css` file where we're using Tailwind - e.g. payments, settings, and now content? That would be imported before/at the compilation step, so all unused styles should get stripped from the final output.

> Another question/suggestion, could we include a root `index.ts` and maybe even `index.css` files in `fxa-react`? This way we import `tailwind.out.css` once, and we are good to go.

I _think_ one potential hiccup with that is we might need to import that root TSX file everywhere to get the styles to load. It works in other packages because those roots are the root of the entire application and `App.tsx` or wherever we have it will _always_ be loaded in no matter what component since the root component is the parent but that's not the case with `fxa-react`.

Reply:
Oops missed one.

> Another question, once we start including custom tailwind utility classes in fxa-react won't we need to include the tailwind.out.css from fxa-react in any case?

If it's a custom utility class where we can modify it in the TW config, I don't think so. If you're mean [component classes](https://github.com/mozilla/fxa/blob/main/packages/fxa-settings/README.md#component-classes) with Tailwind's `@apply`, I think that's kind of the same issue we're trying to solve here. I mentioned this in my previous comment: "I wonder if for `fxa-react`, we could import CSS files directly like importing `portal.css` here instead of `tailwind.out.css`, but the problem with that is when we use `@apply` in CSS files in there, it would need to get compiled first."

I'm hoping we can just import `fxa-react`'s `tailwind.css` file in each of our packages' `tailwind.css` file as I described in my previous comment. Fingers crossed that solves all our problems 😬

Reply:
> After updating package.json to include --postcss in the build-css script,

Heh, why do we need to do this? Is that required in order to `import 'fxa-react/styles/tailwind.css';` to our other `tailwind.css` files?

Wonder if that build error is related to an error I was hitting with Storybook and `postcss-loader`. I was able to tell that package to use post css 8 [here](https://github.com/mozilla/fxa/blob/main/packages/fxa-settings/.storybook/main.js#L16) 🤔

> I was thinking of creating an `index.css` file to collect all other CSS files in `fxa-react`, and then other packages could just import that from `fxa-react`. 

Would this allow us to use Tailwind directives like `@apply` in our CSS? I think it would so long as you're suggesting importing this file into our `tailwind.css` files before they're transformed into `tailwind.out.css` - I'm probably not connecting a couple of the dots here, the `tailwind.css` file right now basically collects those CSS files to how you described, but I know it has TW directives so maybe that's the problem. That seems fine, just not sure why we wouldn't use `tailwind.css` vs this new file. 🙂

Reply:
> If the `--postcss` option isn't added, then the `tailwindcss` command just adds  `@import '../../../fxa-react/styles/index.css';` to `tailwind.out.css`, and leaves all `@apply` and `@tailwind` directives as is. So for `tailwindcss` to actually turn it into plain `css`, we need the `--postcss` option.

Ah gotcha, do all packages in question have `postcss-import`? That's what'll look for `@import` and make the importing happen, so long as you also have the plugin defined in a `postcss.config` file. Looks like at least `fxa-react` and `fxa-payments-server` do - but I don't see that `fxa-react` has a `postcss.config` file, maybe it needs one. Otherwise, `--postcss` seems okay. 😁

Also seems fine then if `@base` is what's giving you those problems. We should have an additional comment somewhere about that, and how we should move those imports into `fxa-react`'s `tailwind.css` and import it from `fxa-react` once we can use `@base` over there.

Yeah I'm open to do a quick call. Let's chat on Slack about the time.